### PR TITLE
feat: make members of SchemaGenerator protected

### PR DESCRIPTION
**Pull request by sramam**
Currently, all non-public members of SchemaGenerator are private.
When customizing SchemaGeneration to fit a project, and/or debugging
the schema generator in the context of the containing project, it is very 
convenient to be able to create a sub-class and extend add necessary
methods. 

Converting all members of SchemaGenerator to protected, provides 
this functionality without compromising the containment intent. 
A buyer-beware foot-gun if you please. 

PS: I'll remove the windows-support modifications if you decide not to accept that change.

### CODE REVIEW

File: test/config.test.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ function assertSchema(
         const validator = new Ajv({
             // skip full check if we are not encoding refs
             validateFormats: config.encodeRefs === false ? undefined : true,
+            allowUnionTypes: true,
```
Comment (domoritz) [inline test/config.test.ts:54]: If this ch age is not necessary, please remove it.

**Review by domoritz — APPROVED** [review]
I'll merge the windows support first. 

Thanks for the pull request.

File: test/config.test.ts (hunk 1 above)
Comment (sramam) [reply test/config.test.ts]: done

**Comment by domoritz** [comment]
Please rebase/merge master

**Comment by sramam** [comment]
> Please rebase/merge master

done

**Pull request merged**